Lib/asyncio/proactor_events.py

```diff
@@ -218,12 +237,20 @@ def resume_reading(self):
         if self._read_fut is None:
             self._loop.call_soon(self._loop_reading, None)
 
-        length = self._pending_data_length
-        self._pending_data_length = -1
-        if length > -1:
-            # Call the protocol methode after calling _loop_reading(),
-            # since the protocol can decide to pause reading again.
-            self._loop.call_soon(self._data_received, self._data[:length], length)
+        if isinstance(self._protocol, protocols.BufferedProtocol):
```

Comment:
I don't see the corresponding changes in selector_events.py. Does it handle these use cases already?

Reply:
Yes it has the following bit of code:
```python
    def set_protocol(self, protocol):
        if isinstance(protocol, protocols.BufferedProtocol):
            self._read_ready_cb = self._read_ready__get_buffer
        else:
            self._read_ready_cb = self._read_ready__data_received

        super().set_protocol(protocol)
```